src/base.js

```diff
@@ -1,95 +1,88 @@
 	search: document.getElementById('search'),
 	searchId: document.getElementById('searchId')
 };
-const COMMON_ERROR = 'An error occurred during getting information!';
-const NOTEXIST_ERROR = 'Current source does not exist!';
+var COMMON_ERROR = 'An error occurred during getting information!';
+var NOTEXIST_ERROR = 'Current source does not exist!';
 
 window.onload = () => {
 	BindAutocomplete(sources);
 	LoadMainPage(sources);
 	addEventListeners();
-}
-
-const addEventListeners = () => {
-
-	document.getElementById('searchBtn').addEventListener('click', function () {
-		LoadArticles(ELEMENTS.searchId.value);
-	});
+};
 
+var addEventListeners = () => {
 
-	document.getElementById('mainPage').addEventListener('click', function () {
-		LoadMainPage(sources);
-	});
-}
+	document.getElementById('searchBtn').addEventListener('click', () => { LoadArticles(ELEMENTS.searchId.value); });
 
-const sources = service.GetAllItems(`${Constants.SERVICE_URL_SOURCES}?apiKey=${Constants.API_KEY}`)
-	.then(data => data.sources);
+	document.getElementById('mainPage').addEventListener('click', () => { LoadMainPage(sources); });
+};
 
+const query = `${Constants.SERVICE_URL_SOURCES}?apiKey=${Constants.API_KEY}`;
+const sources = service.GetAllItems(query).then((data) => {
+	return data.sources;
+});
 
 function LoadMainPage(sources) {
 
-	sources.then(function (data) {
+	sources.then((data) => {
 		htmlHelper.clearContainer(ELEMENTS.container);
-		let top10sources = htmlHelper.getFirstNElements(data, 10);
+		const top10sources = htmlHelper.getFirstNElements(data, 10);
 		top10sources.map(source => {
-			let sourceItem = new SourceItem(source);
-			htmlHelper.appendElement(ELEMENTS.container, sourceItem.getTemplate());
-		});
-	})
-		.catch(function (error) {
-			ELEMENTS.modalBody.innerHTML = COMMON_ERROR;
-			$('#dialog').modal('show');
+			var sourceItem = new SourceItem(source);
+			ELEMENTS.container.appendChild(sourceItem.getTemplate());
 		});
-
+	}).catch((error) => {
+		ELEMENTS.modalBody.innerHTML = COMMON_ERROR;
+		console.log(error.get_message()); //test plugin
+		$('#dialog').modal('show');
+	});
 }
 
 function LoadArticles(sourceId) {
-
+    const query = Constants.SERVICE_URL_ARTICLES + '?sources=' + sourceId + '&apiKey=' + Constants.API_KEY;
 	if (sourceId) {
-		service.GetAllItems(`${Constants.SERVICE_URL_ARTICLES}?sources=${sourceId}&apiKey=${Constants.API_KEY}`)
-			.then(function (data) {
-				htmlHelper.clearContainer(ELEMENTS.container);
-				let articles = data.articles;
-				articles.map(article => {
-					let articleItem = new ArticleItem(article);
-					htmlHelper.appendElement(ELEMENTS.container, articleItem.getTemplate());
-				});
-			})
-			.catch(function (error) {
-				ELEMENTS.modalBody.innerHTML = COMMON_ERROR;
-				$('#dialog').modal('show');
+		service.GetAllItems(query).then((data) => {
+			htmlHelper.clearContainer(ELEMENTS.container);
+			const articles = data.articles;
+			articles.map(article => {

```

Comment:
We have discussion by prev PR,
Why did you use map as loop?
Use `forEach`, please
`https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Array/map`
`https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach`

Reply:
done